src/components/ProjectSettingsTabs/PluginsTab/getFormSchema.ts

```diff
@@ -0,0 +1,182 @@
+            title: "Ticket Search Projects",
+            items: {
+              type: "object" as "object",
+              properties: {
+                searchProject: {
+                  type: "string" as "string",
+                  title: "Search Project",
+                },
+              },
+            },
+          },
+          customTicket: {
+            type: ["boolean", "null"],
+            oneOf: radioBoxOptions(
+              ["Custom Ticket Creation", "Build Baron Ticket Creation"],
+              undefined
+            ),
+          },
+          ticketCreateProject: {
+            type: ["string", "null"],
+            title: "Ticket Create Project",
+          },
+          taskAnnotationSettings: {
+            title: "Custom Ticket Creation",
+            type: "object" as "object",
+            properties: {
+              fileTicketWebhook: {
+                type: "object" as "object",
+                title: "Webhook",
+                properties: {
+                  endpoint: {
+                    type: ["string", "null"],
+                    title: "Endpoint",
+                  },
+                  secret: {
+                    type: ["string", "null"],
+                    title: "Secret",
+                  },
+                },
+              },
+              jiraCustomFields: {
+                type: "array" as "array",
+                title: "Custom Jira Fields",
+                items: {
+                  type: "object" as "object",
+                  properties: {
+                    field: {
+                      type: "string" as "string",
+                      title: "Field",
+                    },
+                    displayText: {
+                      type: "string" as "string",
+                      title: "Display Text",
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+  uiSchema: {
+    performanceSettings: {
+      "ui:rootFieldId": "plugins",
+      "ui:ObjectFieldTemplate": CardFieldTemplate,
+      perfEnabled: {
+        "ui:widget": widgets.RadioBoxWidget,
+      },
+    },
+    buildBaronSettings: {
+      "ui:description":
+        "Define either a webhook for custom build failure ticket creation, or specify existing JIRA projects.",
+      "ui:rootFieldId": "buildBaron",
+      "ui:ObjectFieldTemplate": CardFieldTemplate,
+      ticketSearchProjects: {
+        ...placeholderIf(
+          repoData?.projectPluginsSettings?.buildBaronSettings
+            ?.ticketSearchProjects
+        ),
+        "ui:buttonText": "Add Search Project",
+        options: {
+          useRepoSettings,
+        },
+      },
+      customTicket: {
+        "ui:widget": widgets.RadioBoxWidget,
+        "ui:showLabel": false,
+        "ui:data-cy": "enabled-radio-box",
+      },
+      ticketCreateProject: {
+        "ui:disabled": formData?.buildBaronSettings.customTicket === true,
```

Comment:
What do you think about using LeafyGreen's [segmented control](https://www.mongodb.design/component/segmented-control/example/) component to display the correct fields based on whether a user has chosen "custom ticket creation" or "build baron ticket creation"? I'm using it right now for some fields on the GitHub settings page and while it requires more logic on our end, I think it's a pretty effective pattern for displaying form fields like this. Interested in your thoughts!

Reply:
In the linked example, nothing changes when I toggle to a different fruit... hmm...

Reply:
This doesn't look very different than the radio buttons/toggles we are currently using. How would using this component change things?

Reply:
![custom](https://user-images.githubusercontent.com/13104717/149144382-b78ef05a-72d0-48a0-bf45-f03d51b1b104.gif)
)

Reply:
I think design was pretty against having fields be hidden based on toggles so we may want to run this by them.

Reply:
@sophstad Since you plan to use this design pattern as well, should we bring it up at the next design sync?

Reply:
Sure, although the use case for alias definitions is a little different and use of a toggle is already established in their designs. We should probably ask about this page specifically!

Reply:
sounds good, I'll add it to the agenda

Reply:
@malikchaya2 should custom jira fields only be available for JIRA ticket creation?

Reply:
Gotcha; Can Custom Jira Fields and Ticket Search Project be together then, since they're both configurable regardless of the toggle? (either at the top or bottom)

Reply:
> Gotcha; Can Custom Jira Fields and Ticket Search Project be together then, since they're both configurable regardless of the toggle? (either at the top or bottom)

@malikchaya2 can this be addressed?

Reply:
It is set up for assigned teams which is what they originally wanted the custom fields for. In order to populate assigned teams, it would need to be added as a custom field. 
https://github.com/evergreen-ci/spruce/blob/main/src/pages/task/taskTabs/buildBaronAndAnnotations/BBComponents.tsx#L88

Reply:
I thought about that, but decided against it because it will probably only be used by one team. I didn't want it to be too visible. Thoughts?

Reply:
I think it's okay to move even if most teams won't use it; if anything it's more visible where it is because users need to consider whether or not it's required for Custom Ticket / JIRA ticket creation.

Reply:
that's a good point, I'll move it

Reply:
I realized that you can't actually use `ticket search project` with custom ticket creation, so I will move that over.